examples/timelockmultisig.sol

```diff
@@ -0,0 +1,250 @@
+            m.canVote = canVote;
+        }
+
+        MembershipChanged(targetMember, canVote);
+
+    }
+
+    function changeMembers(address[] newMembers, bool canVote) {
+        for (uint i = 0; i < newMembers.length; i++) {
+            changeMembership(newMembers[i], canVote, '');
+        }
+    }
+
+    /* Function to create a new proposal */
+    function newProposal(
+        address beneficiary,
+        uint weiAmount,
+        string jobDescription,
+        bytes transactionBytecode
+    )
+        onlyMembers
+        returns (uint proposalID)
+    {
+        proposalID = proposals.length++;
+        Proposal p = proposals[proposalID];
+        p.recipient = beneficiary;
+        p.amount = weiAmount;
+        p.description = jobDescription;
+        p.proposalHash = sha3(beneficiary, weiAmount, transactionBytecode);
+        p.executed = false;
+        ProposalAdded(proposalID, beneficiary, weiAmount, jobDescription);
+        numProposals = proposalID+1;
+        vote(proposalID, true, '');
+
+        return proposalID;
+    }
+
+
+    /* Function to create a new proposal */
+    function newProposalInEther(
+        address beneficiary,
+        uint etherAmount,
+        string jobDescription,
+        bytes transactionBytecode
+    )
+        onlyMembers
+        returns (uint proposalID)
+    {
+        return newProposal(beneficiary, etherAmount * 1 ether, jobDescription, transactionBytecode);
+    }
+
+    /* function to check if a proposal code matches */
+    function checkProposalCode(
+        uint proposalNumber,
+        address beneficiary,
+        uint etherAmount,
+        bytes transactionBytecode
+    )
+        constant
+        returns (bool codeChecksOut)
+    {
+        Proposal p = proposals[proposalNumber];
+        return p.proposalHash == sha3(beneficiary, etherAmount, transactionBytecode);
+    }
+
+    function vote(
+        uint proposalNumber,
+        bool supportsProposal,
+        string justificationText
+    )
+        onlyMembers
+        returns (uint voteID)
+    {
+        Proposal p = proposals[proposalNumber];         // Get the proposal
+        if (p.voted[msg.sender] == true) throw;         // If has already voted, cancel
+        p.voted[msg.sender] = true;                     // Set this voter as having voted
+        if (supportsProposal) {                         // If they support the proposal
+            p.currentResult++;                          // Increase score
+        } else {                                        // If they don't
+            p.currentResult--;                          // Decrease the score
+        }
+        // Create a log of this event
+        Voted(proposalNumber,  supportsProposal, msg.sender, justificationText);
+    }
+
+    function proposalDeadline(uint proposalNumber) constant returns(uint deadline) {
+        Proposal p = proposals[proposalNumber];
+        uint factor = 3*10**uint(6 - (5 * p.currentResult / int(members.length - 1)));
```

Comment:
It seems like there's a risk of dividing by zero here. Or is that prevented by line 115?